src/modules/user/user.service.ts

```diff
@@ -42,27 +43,39 @@ export class UserService {
 
   // get all users
   async findAll(): Promise<User[]> {
-    const res = await this.userRepository.find();
-    return res;
+    const users = await this.userRepository.find();
+    users.forEach(user => {
+      if (user.lastLoginTime && user.timezone) {
+        user.lastLoginTime = convertToLocalTime(
+          user.lastLoginTime,
+          user.timezone,
+        );
+      }
+    });
+    return users;
   }
 
   // find an user by id
   async find(id: string): Promise<User> {
-    const res = await this.userRepository.findOne({
-      where: {
-        id,
-      },
-    });
-    return res;
+    const user = await this.userRepository.findOne({ where: { id } });
+    if (user && user.lastLoginTime && user.timezone) {
```

Comment:
the last login time and timezone won't be changed after user queries being executed